libraries/Microsoft.Bot.Builder.AI.LuisV3/LuisUtil.cs

```diff
@@ -0,0 +1,194 @@
+                        path = path.Substring(0, path.LastIndexOf('.'));
+                        geoTypes.Add(path, type.Value<string>().Substring(GeoV2.Length));
+                    }
+                    else
+                    {
+                        foreach (var property in obj.Properties())
+                        {
+                            FindGeographyTypes(property.Value, geoTypes);
+                        }
+                    }
+                }
+                else if (source is JArray arr)
+                {
+                    foreach (var elt in arr)
+                    {
+                        FindGeographyTypes(elt, geoTypes);
+                    }
+                }
+            }
+        }
+
+        internal static JToken MapProperties(JToken source, bool inInstance, Dictionary<string, string> geoTypes)
+        {
+            var result = source;
+            if (source is JObject obj)
+            {
+                var nobj = new JObject();
+
+                // Fix datetime by reverting to simple timex
+                if (!inInstance && obj.TryGetValue("type", out var type) && type.Type == JTokenType.String && _dateSubtypes.Contains(type.Value<string>()))
+                {
+                    var timexs = obj["values"];
+                    var arr = new JArray();
+                    if (timexs != null)
+                    {
+                        var unique = new HashSet<string>();
+                        foreach (var elt in timexs)
+                        {
+                            unique.Add(elt["timex"]?.Value<string>());
+                        }
+
+                        foreach (var timex in unique)
+                        {
+                            arr.Add(timex);
+                        }
+
+                        nobj["timex"] = arr;
+                    }
+
+                    nobj["type"] = type;
+                }
+                else
+                {
+                    // Map or remove properties
+                    foreach (var property in obj.Properties())
+                    {
+                        var name = NormalizeEntity(property.Name);
+                        var isObj = property.Value.Type == JTokenType.Object;
+                        var isArr = property.Value.Type == JTokenType.Array;
+                        var isStr = property.Value.Type == JTokenType.String;
+                        var isInt = property.Value.Type == JTokenType.Integer;
+                        var val = MapProperties(property.Value, inInstance || property.Name == MetadataKey, geoTypes);
+                        if (name == "datetime" && isArr)
+                        {
+                            nobj.Add("datetimeV1", val);
+                        }
+                        else if (name == "datetimeV2" && isArr)
+                        {
+                            nobj.Add("datetime", val);
+                        }
+                        else if (inInstance)
```

Comment:
Where's ordinalV1?

Reply:
The V3 endpoint does not surface ordinalV1--even if part of the model.